### PR TITLE
Add deep ensemble split utility

### PR DESCRIPTION
**Pull request by Kirscher**
This PR adds a small opt-in utility for training deep ensemble members using nnU-Net’s existing fold infrastructure.

It does not change the training loop, architecture, loss functions, checkpoint format, inference defaults, or model-selection defaults. The new command only creates or updates `splits_final.json` by preserving existing CV folds and appending marked full-training-set folds for deep ensemble training.

Users can then train those members with the existing command:

```bash
nnUNetv2_train DATASET CONFIGURATION FOLD
```

and explicitly select them for inference with `nnUNetv2_predict -f`.

The PR also adds documentation explaining the difference between CV ensembles and deep ensembles for uncertainty estimation, warns that validation metrics from full-training-set deep ensemble folds are biased, and includes lightweight tests for split generation behavior.

### CODE REVIEW

File: nnunetv2/evaluation/find_best_configuration.py [hunk 1]
```diff
@@ -24,6 +25,20 @@
 ])
 
 
+def maybe_warn_about_deep_ensemble_folds(dataset_name_or_id: Union[str, int], folds: Union[List[int], Tuple[int, ...]]):
+    splits_file = join(nnUNet_preprocessed, maybe_convert_to_dataset_name(dataset_name_or_id), 'splits_final.json')
```
[the remark this answers is not in the document]
Comment (FabianIsensee) [reply nnunetv2/evaluation/find_best_configuration.py]: @Kirscher can you take a look?

File: nnunetv2/utilities/deep_ensemble_splits.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    from batchgenerators.utilities.file_and_folder_operations import isfile, join, load_json, save_json
+
+    from nnunetv2.utilities.dataset_name_id_conversion import maybe_convert_to_dataset_name
+
+    if num_members < 1:
+        raise ValueError("num_members must be >= 1.")
+
+    dataset_name = maybe_convert_to_dataset_name(dataset_name_or_id)
+    preprocessed_dataset_folder, preprocessed_configuration_folder = _get_preprocessed_configuration_folder(
+        dataset_name, plans_identifier, configuration)
```
[the remark this answers is not in the document]
Comment (FabianIsensee) [reply nnunetv2/utilities/deep_ensemble_splits.py]: @Kirscher
